Foundation/NSGeometry.swift

```diff
@@ -91,6 +95,61 @@ public func ==(lhs: CGPoint, rhs: CGPoint) -> Bool {
     return lhs.x == rhs.x && lhs.y == rhs.y
 }
 
+extension CGPoint: NSSpecialValueCoding {
+    init(bytes: UnsafePointer<Void>) {
+        let buffer = UnsafePointer<CGFloat>(bytes)
+
+        self.x = buffer.memory
+        self.y = buffer.advancedBy(1).memory
+    }
+    
+    public init?(coder aDecoder: NSCoder) {
```

Comment:
These are API changes; id rather not have these be public if they are needed to make the special values work since it will mean that there is disparity; now perhaps with a later change we might want to consider this a first step for value-type coding but that should be considered  separately from the functionality of NSCoding